packages/material-ui-styles/src/ThemeProvider/nested.js

```diff
@@ -1,3 +1,3 @@
-const nested = Symbol('nested');
+const hasSymbol = typeof Symbol === 'function';
 
-export default nested;
+export default (hasSymbol ? Symbol.for('nested') : '__THEME_NESTED__');
```

Comment:
```suggestion
export default (hasSymbol ? Symbol.for('mui.nested') : '__THEME_NESTED__');
```

`Symbol.for` is global. It can collide with other symbols which is why we want to namespace it.